terminal/terminal.md

```diff
@@ -0,0 +1,417 @@
+```
+We can scroll down with the key arrows and quit **q**
+
+Whenever you don't know how to use a command use man then the name of the command to read everything about it, it also shows the flags you can use and in some cases some examples.
+
+But now what if only want a small description? Well we are all always in a hurry so let's be more pragmatic and use the **whatis** command
+
+```
+vagrant@vagrant-ubuntu-trusty-64:~$ whatis ls
+ls (1)               - list directory contents
+```
+pretty neat
+
+What about the other way? I want a command if there is a command that lists the directory contents... if only...
+```
+vagrant@vagrant-ubuntu-trusty-64:~$ apropos "list directory contents"
+```
+```
+vagrant@vagrant-ubuntu-trusty-64:~$ apropos "list directory contents"
+dir (1)              - list directory contents
+ls (1)               - list directory contents
+ntfsls (8)           - list directory contents on an NTFS filesystem
+vdir (1)             - list directory contents
+```
+Yes, the command **apropos** searches for a command that fits your description
+
+#### Creating files and directories
+
+We have been exploring how to move across directories, list files and see the metadata of those, but now, let's go one step ahead and create a directory with the **mkdir** command
+
+```
+danielvilla@Daniels-MacBook-Air:~$ mkdir mydirectory
+danielvilla@Daniels-MacBook-Air:~$ 
+```
+Normally when a command is successful it won't print anything, if we look now the mydirectory directory is created and ready to have files
+
+It's very common that we want to create directories contained inside directories (that's how everything works) but mkdir alone won't let us do that
+
+```
+danielvilla@Daniels-MacBook-Air:~$ mkdir great_ideas/best_ideas/dontlookhere
+mkdir: great_ideas/best_deas: No such file or directory
+```
+We need to add the **p** flag to correct that
+
+```
+danielvilla@Daniels-MacBook-Air:~$ mkdir -p great_ideas/best_ideas/dontlookhere
+danielvilla@Daniels-MacBook-Air:~$ 
+```
+That will save us a lot of time of doing `mkdir great_ideas` then `cd great_ideas` then `mkdir best_ideas`... well you get the point
+
+NOW! we have our directory ready to have files and great ideas, but how do we create a simple file? 
+
+There are plenty of ways to create files on unix system but the common ones are
+
+```
+danielvilla@Daniels-MacBook-Air:~$ touch idea1.txt
+```
+**touch** is an awesome command, you don't have to worry if the file exists and if it is going to get overwritten, if there's a idea1.txt file only the created date wil be affected to the current date, the data will stay intact
+
+
+**vim** is a text editor that will let you create files, the comprehension and usage of vim is WAY beyond this course, but I want to let you know people who know how to use vim are VERY efficient in matters of text insertion and deletion, vim alows you to do very delicated and elegant modifications to a file. For now lets just create a file
```

Comment:
**vim** is a text editor that will let you create files, the comprehension and usage of vim is WAY beyond this course, but I want to let you know `that` people who know how to use vim are VERY efficient in matters of text insertion and deletion, vim al`l`ows you to do very delicat~ed~ and elegant modifications to a file. For now lets just create a file`:`